Controller/GroupsController.php

```diff
@@ -39,19 +43,32 @@ public function getConfig()
 
             if (count($groupWithSuchRefNameList) > 0) {
                 $groupWithSuchRefName = $groupWithSuchRefNameList[0];
-                $validationResult->addFieldError(
-                    'refName',
-                    T::trans(
-                        'This refName is taken. Consider use \'%groupName%\' group or change current reference name.',
-                        array('%groupName%' => $groupWithSuchRefName->getName())
-                    )
-                );
+
+                if ($groupWithSuchRefName->getId() != $group->getId()) {
+                    $validationResult->addFieldError(
+                        'refName',
+                        T::trans(
+                            'This refName is taken. Consider use \'%groupName%\' group or change current reference name.',
+                            array('%groupName%' => $groupWithSuchRefName->getName())
+                        )
+                    );
+                }
             }
 
             return $validationResult;
 
         };
 
+        $mapEntity = function (array $params, $group, DataMapperInterface $defaultMapper, ContainerInterface $container) {
+            $group->setName($params['name']);
```

Comment:
It is better to invoke here $defaultMapper and let it do this for you, later if new fields are added then no extra work need to be done.